diff -x '*~' -x '.*' -x '*.orig' -x '*.rej' --speed-large-files --minimal -Nru tmp/aMule-2.3.1/configure work/main.d/aMule-2.3.1/configure
--- tmp/aMule-2.3.1/configure	2011-11-11 17:59:26.000000000 -0300
+++ new-aMule-2.3.1/configure	2012-09-16 20:11:04.601299959 -0300
@@ -8569,7 +8569,7 @@
 fi
 
 
-	if test $cross_compiling = no; then :
+	if test $cross_compiling = xno; then :
 
 		{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for libupnp version >= 1.6.6" >&5
 $as_echo_n "checking for libupnp version >= 1.6.6... " >&6; }
