--- old/Makefile	2011-07-27 11:52:56.062293053 -0300
+++ new/Makefile	2011-07-26 20:29:27.343701012 -0300
@@ -4,7 +4,7 @@
 MASTER_SITES  = http://ftp.gnu.org/gnu/$(GARNAME)/
 MASTER_SITES += ftp://sources.redhat.com/pub/$(GARNAME)/snapshots/
 DISTFILES = $(DISTNAME).tar.bz2
-PATCHFILES = $(DISTNAME)-select_fd_type.patch $(DISTNAME)-__i686.patch $(DISTNAME)-build_lib.patch $(DISTNAME)-build_locale.patch $(DISTNAME)-build_sunrpc.patch $(DISTNAME)-perl.patch.gar $(DISTNAME).patch $(DISTNAME)-ldd.patch
+PATCHFILES = $(DISTNAME)-rpc_key_prot_h.patch $(DISTNAME)-select_fd_type.patch $(DISTNAME)-__i686.patch $(DISTNAME)-build_lib.patch $(DISTNAME)-build_locale.patch $(DISTNAME)-build_sunrpc.patch $(DISTNAME)-perl.patch.gar $(DISTNAME).patch $(DISTNAME)-ldd.patch
 LICENSE = GPL2/LGPL2_1
 
 DESCRIPTION = 
