diff -Naur MPlayer-29438-old/etc/codecs.conf MPlayer-29438-new/etc/codecs.conf
--- MPlayer-29438-old/etc/codecs.conf	2009-07-30 10:36:18.000000000 -0700
+++ MPlayer-29438-new/etc/codecs.conf	2009-07-30 10:43:36.000000000 -0700
@@ -3553,6 +3553,13 @@
   fourcc "MLP " ; internal MPlayer FourCC
   driver ffmpeg
   dll "mlp"
+  
+audiocodec fftruehd
+  info "FFmpeg TrueHD"
+  status working
+  fourcc TRHD ; internal MPlayer FourCC
+  driver ffmpeg
+  dll "truehd"
 
 audiocodec ffnellymoser
   info "FFmpeg Nellymoser Audio"
diff -Naur MPlayer-29438-old/libmpdemux/demux_ts.c MPlayer-29438-new/libmpdemux/demux_ts.c
--- MPlayer-29438-old/libmpdemux/demux_ts.c	2009-07-24 09:24:15.000000000 -0700
+++ MPlayer-29438-new/libmpdemux/demux_ts.c	2009-07-30 10:43:36.000000000 -0700
@@ -77,6 +77,7 @@
 	AUDIO_DTS	= 0x2001,
 	AUDIO_LPCM_BE  	= 0x10001,
 	AUDIO_AAC	= mmioFOURCC('M', 'P', '4', 'A'),
+	AUDIO_TRUEHD	= mmioFOURCC('T', 'R', 'H', 'D'),
 	SPU_DVD		= 0x3000000,
 	SPU_DVB		= 0x3000001,
 	PES_PRIVATE1	= 0xBD00000,
@@ -244,7 +245,7 @@
 } TS_pids_t;
 
 
-#define IS_AUDIO(x) (((x) == AUDIO_MP2) || ((x) == AUDIO_A52) || ((x) == AUDIO_LPCM_BE) || ((x) == AUDIO_AAC) || ((x) == AUDIO_DTS))
+#define IS_AUDIO(x) (((x) == AUDIO_MP2) || ((x) == AUDIO_A52) || ((x) == AUDIO_LPCM_BE) || ((x) == AUDIO_AAC) || ((x) == AUDIO_DTS) || ((x) == AUDIO_TRUEHD))
 #define IS_VIDEO(x) (((x) == VIDEO_MPEG1) || ((x) == VIDEO_MPEG2) || ((x) == VIDEO_MPEG4) || ((x) == VIDEO_H264) || ((x) == VIDEO_AVC)  || ((x) == VIDEO_VC1))
 
 static int ts_parse(demuxer_t *demuxer, ES_stream_t *es, unsigned char *packet, int probe);
@@ -871,6 +872,8 @@
 		mp_msg(MSGT_DEMUXER, MSGL_INFO, "AUDIO LPCM(pid=%d)", param->apid);
 	else if(param->atype == AUDIO_AAC)
 		mp_msg(MSGT_DEMUXER, MSGL_INFO, "AUDIO AAC(pid=%d)", param->apid);
+	else if(param->atype == AUDIO_TRUEHD)
+		mp_msg(MSGT_DEMUXER, MSGL_INFO, "AUDIO TRUEHD(pid=%d)", param->apid);
 	else
 	{
 		audio_found = 0;
@@ -1410,8 +1413,11 @@
 	if(stream_id==0xfd)
 	{
 		int ssid = parse_pes_extension_fields(p, pkt_len);
+		//fprintf(stderr, "\r\nssid: %d, 0x%x\r\n", ssid, ssid);
 		if((audio_substream_id!=-1) && (ssid != audio_substream_id))
 			return 0;
+		if(ssid == 0x72)
+			es->type  = type_from_pmt = AUDIO_TRUEHD;
 	}
 
 	p += header_len + 9;
diff -Naur MPlayer-29438-old/libmpdemux/mp_taglists.c MPlayer-29438-new/libmpdemux/mp_taglists.c
--- MPlayer-29438-old/libmpdemux/mp_taglists.c	2009-07-24 09:24:15.000000000 -0700
+++ MPlayer-29438-new/libmpdemux/mp_taglists.c	2009-07-30 10:43:36.000000000 -0700
@@ -36,6 +36,7 @@
     { CODEC_ID_EAC3,              MKTAG('E', 'A', 'C', '3')},
     { CODEC_ID_INTERPLAY_DPCM,    MKTAG('I', 'N', 'P', 'A')},
     { CODEC_ID_MLP,               MKTAG('M', 'L', 'P', ' ')},
+    { CODEC_ID_TRUEHD,            MKTAG('T', 'R', 'H', 'D')},
     { CODEC_ID_MUSEPACK7,         MKTAG('M', 'P', 'C', ' ')},
     { CODEC_ID_MUSEPACK8,         MKTAG('M', 'P', 'C', '8')},
     { CODEC_ID_NELLYMOSER,        MKTAG('N', 'E', 'L', 'L')},
