diff -Naur MPlayer-1.0rc2-old/configure MPlayer-1.0rc2-new/configure
--- MPlayer-1.0rc2-old/configure	2008-09-27 08:04:03.000000000 -0700
+++ MPlayer-1.0rc2-new/configure	2008-09-27 08:16:45.000000000 -0700
@@ -3857,21 +3857,21 @@
 echocheck "X11 headers presence"
   _x11_headers="no"
   _res_comment="check if the dev(el) packages are installed"
-  for I in `echo $_inc_extra | sed s/-I//g` /usr/include ; do
-    if test -f "$I/X11/Xlib.h" ; then
+#  for I in `echo $_inc_extra | sed s/-I//g` /usr/include ; do
+#    if test -f "$I/X11/Xlib.h" ; then
       _x11_headers="yes"
       _res_comment=""
-      break
-    fi
-  done
-  for I in /usr/X11/include /usr/X11R6/include /usr/include/X11R6 /usr/openwin/include ; do
-    if test -f "$I/X11/Xlib.h" ; then
-      _inc_extra="$_inc_extra -I$I"
-      _x11_headers="yes"
-      _res_comment="using $I"
-      break
-    fi
-  done
+#      break
+#    fi
+#  done
+#  for I in /usr/X11/include /usr/X11R6/include /usr/include/X11R6 /usr/openwin/include ; do
+#    if test -f "$I/X11/Xlib.h" ; then
+#      _inc_extra="$_inc_extra -I$I"
+#      _x11_headers="yes"
+#      _res_comment="using $I"
+#      break
+#    fi
+#  done
 echores "$_x11_headers"
 
 
