diff -Naur MPlayer-1.0rc2-old/dvdread/dvd_reader.c MPlayer-1.0rc2-new/dvdread/dvd_reader.c
--- MPlayer-1.0rc2-old/dvdread/dvd_reader.c	2007-10-07 12:49:28.000000000 -0700
+++ MPlayer-1.0rc2-new/dvdread/dvd_reader.c	2008-10-26 13:16:49.000000000 -0700
@@ -1390,6 +1390,28 @@
   return offset;
 }
 
+int DVDFileSeekForce( dvd_file_t *dvd_file, int offset, int force_size )
+{
+  /* Check arguments. */
+  if( dvd_file == NULL || offset < 0 )
+    return -1;
+
+  if( dvd_file->dvd->isImageFile ) {
+    if( force_size < 0 )
+      force_size = (offset - 1) / DVD_VIDEO_LB_LEN + 1;
+    if( dvd_file->filesize < force_size ) {
+      dvd_file->filesize = force_size;
+      fprintf(stderr, "libdvdread: Ignored UDF provided size of file.\n");
+    }
+  }
+
+  if( offset > dvd_file->filesize * DVD_VIDEO_LB_LEN ) {
+    return -1;
+  }
+  dvd_file->seek_pos = (uint32_t) offset;
+  return offset;
+}
+
 #ifndef HAVE_UINTPTR_T
 #warning "Assuming that (unsigned long) can hold (void *)"
 typedef unsigned long uintptr_t;
@@ -1441,7 +1463,7 @@
   memcpy( data, &(secbuf[ seek_byte ]), byte_size );
   free( secbuf_start );
 
-  dvd_file->seek_pos += byte_size;
+  DVDFileSeekForce(dvd_file, dvd_file->seek_pos + byte_size, -1);
   return byte_size;
 }
 
diff -Naur MPlayer-1.0rc2-old/dvdread/dvd_reader.h MPlayer-1.0rc2-new/dvdread/dvd_reader.h
--- MPlayer-1.0rc2-old/dvdread/dvd_reader.h	2007-10-07 12:49:28.000000000 -0700
+++ MPlayer-1.0rc2-new/dvdread/dvd_reader.h	2008-10-26 13:16:49.000000000 -0700
@@ -240,6 +240,8 @@
  */
 int DVDFileSeek( dvd_file_t *, int );
 
+int DVDFileSeekForce( dvd_file_t *, int, int );
+
 /**
  * Reads the given number of bytes from the file.  This call can only be used
  * on the information files, and may not be used for reading from a VOB.  This
diff -Naur MPlayer-1.0rc2-old/dvdread/ifo_read.c MPlayer-1.0rc2-new/dvdread/ifo_read.c
--- MPlayer-1.0rc2-old/dvdread/ifo_read.c	2007-10-07 12:49:28.000000000 -0700
+++ MPlayer-1.0rc2-new/dvdread/ifo_read.c	2008-10-26 13:16:49.000000000 -0700
@@ -110,6 +110,10 @@
   return (DVDFileSeek(dvd_file, (int)offset) == (int)offset);
 }
 
+static inline int DVDFileSeekForce_( dvd_file_t *dvd_file, uint32_t offset, int force_size ) {
+  return (DVDFileSeekForce(dvd_file, (int)offset, force_size) == (int)offset);
+}
+
 
 ifo_handle_t *ifoOpen(dvd_reader_t *dvd, int title) {
   ifo_handle_t *ifofile;
@@ -1659,7 +1663,7 @@
   unsigned int i;
   int info_length;
 
-  if(!DVDFileSeek_(ifofile->file, sector * DVD_BLOCK_LEN))
+  if(!DVDFileSeekForce_(ifofile->file, sector * DVD_BLOCK_LEN, sector))
     return 0;
 
   if(!(DVDReadBytes(ifofile->file, vobu_admap, VOBU_ADMAP_SIZE)))
