--- old/Makefile	2011-07-28 11:31:27.848390460 -0300
+++ new/Makefile	2010-10-01 17:14:16.000000000 -0300
@@ -3,7 +3,7 @@
 CATEGORIES = devel
 MASTER_SITES = http://ftp.gnu.org/gnu/$(GARNAME)/
 DISTFILES = $(DISTNAME).tar.gz
-PATCHFILES = $(DISTNAME)-non_interactive.patch $(DISTNAME)-perl.patch.gar $(DISTNAME)-withoutman.patch
+PATCHFILES = $(DISTNAME)-non_interactive.patch $(DISTNAME)-perl.patch.gar
 LICENSE = GPL3
 
 DESCRIPTION = 
@@ -18,6 +18,7 @@
 
 CONFIGURE_ARGS = $(DIRPATHS) --build=$(GARBUILD) --host=$(GARHOST) \
 	--enable-shared \
+	--disable-static \
 	--disable-rpath \
 	--with-gnu-ld
 
--- old/checksums	2011-07-27 20:47:44.332294002 -0300
+++ new/checksums	2010-08-21 10:29:19.000000000 -0300
@@ -1,4 +1,3 @@
 3dd55b952826d2b32f51308f2f91aa89  download/gettext-0.18.1.1.tar.gz
 f4cf79ae3b33015dc4304c72e5cf9604  download/gettext-0.18.1.1-non_interactive.patch
 db239dbf1a6ad795616a879630579cb3  download/gettext-0.18.1.1-perl.patch.gar
-1ec83a03f6110514af9f83755acd7d91  download/gettext-0.18.1.1-withoutman.patch
