diff -Naur xorg-server-1.8.2-old/hw/xfree86/common/xf86AutoConfig.c xorg-server-1.8.2-new/hw/xfree86/common/xf86AutoConfig.c
--- xorg-server-1.8.2-old/hw/xfree86/common/xf86AutoConfig.c	2010-07-01 10:36:59.000000000 -0700
+++ xorg-server-1.8.2-new/hw/xfree86/common/xf86AutoConfig.c	2010-07-01 10:37:23.000000000 -0700
@@ -176,7 +176,7 @@
 	case 0x1142:		    driverList[0] = "apm"; break;
 	case 0xedd8:		    driverList[0] = "ark"; break;
 	case 0x1a03:		    driverList[0] = "ast"; break;
-	case 0x1002:		    driverList[0] = "ati"; break;
+	case 0x1002:		    driverList[0] = "radeon"; break;
 	case 0x102c:		    driverList[0] = "chips"; break;
 	case 0x1013:		    driverList[0] = "cirrus"; break;
 	case 0x3d3d:		    driverList[0] = "glint"; break;
