--- old/ext/dba/config.m4	2008-12-15 18:22:27.000000000 +0100
+++ new/ext/dba/config.m4.oden	2008-12-15 18:45:21.000000000 +0100
@@ -277,11 +277,15 @@
 if test "$PHP_DB4" != "no"; then
   PHP_DBA_STD_BEGIN
   dbdp="/usr/local/BerkeleyDB.4."
-  for i in $PHP_DB4 ${dbdp}6 ${dbdp}5 ${dbdp}4 ${dbdp}3  ${dbdp}2 ${dbdp}1 ${dbdp}0 /usr/local /usr; do
+  for i in $PHP_DB4 ${dbdp}7 ${dbdp}6 ${dbdp}5 ${dbdp}4 ${dbdp}3  ${dbdp}2 ${dbdp}1 ${dbdp}0 /usr/local /usr; do
     if test -f "$i/db4/db.h"; then
       THIS_PREFIX=$i
       THIS_INCLUDE=$i/db4/db.h
       break
+    elif test -f "$i/include/db4.7/db.h"; then
+      THIS_PREFIX=$i
+      THIS_INCLUDE=$i/include/db4.7/db.h
+      break
     elif test -f "$i/include/db4.6/db.h"; then
       THIS_PREFIX=$i
       THIS_INCLUDE=$i/include/db4.6/db.h
@@ -308,7 +312,7 @@
       break
     fi
   done
-  PHP_DBA_DB_CHECK(4, db-4.6 db-4.5 db-4.4 db-4.3 db-4.2 db-4.1 db-4.0 db-4 db4 db, [(void)db_create((DB**)0, (DB_ENV*)0, 0)])
+  PHP_DBA_DB_CHECK(4, db-4.7 db-4.6 db-4.5 db-4.4 db-4.3 db-4.2 db-4.1 db-4.0 db-4 db4 db, [(void)db_create((DB**)0, (DB_ENV*)0, 0)])
 fi
 PHP_DBA_STD_RESULT(db4,Berkeley DB4)
 
