--- old/ext/xmlrpc/config.m4	2007-06-11 11:48:59.000000000 +0200
+++ new/ext/xmlrpc/config.m4.oden	2007-06-11 12:21:28.000000000 +0200
@@ -49,7 +49,7 @@
     for i in $PHP_LIBEXPAT_DIR $XMLRPC_DIR /usr/local /usr; do
       if test -f $i/$PHP_LIBDIR/libexpat.a || test -f $i/$PHP_LIBDIR/libexpat.$SHLIB_SUFFIX_NAME; then
         AC_DEFINE(HAVE_LIBEXPAT,1,[ ])
-        PHP_ADD_LIBRARY_WITH_PATH(expat, $i/$PHP_LIBDIR, XMLRPC_SHARED_LIBADD)
+        PHP_ADD_LIBRARY_WITH_PATH(expat,,XMLRPC_SHARED_LIBADD)
         PHP_ADD_INCLUDE($i/include)
         testval=yes
         break
@@ -100,7 +100,8 @@
   fi
 
   PHP_ADD_INCLUDE($XMLRPC_DIR)
-  PHP_ADD_LIBRARY_WITH_PATH(xmlrpc, $XMLRPC_DIR/$PHP_LIBDIR, XMLRPC_SHARED_LIBADD)
+  PHP_ADD_LIBRARY_WITH_PATH(xmlrpc,,XMLRPC_SHARED_LIBADD)
   PHP_NEW_EXTENSION(xmlrpc,xmlrpc-epi-php.c, $ext_shared)
+  PHP_SUBST(XMLRPC_SHARED_LIBADD)
   XMLRPC_MODULE_TYPE=external
 fi
