diff -Naur perl-5.12.2-old/ext/Errno/Errno_pm.PL perl-5.12.2-new/ext/Errno/Errno_pm.PL
--- perl-5.12.2-old/ext/Errno/Errno_pm.PL	2010-09-15 10:41:49.000000000 -0700
+++ perl-5.12.2-new/ext/Errno/Errno_pm.PL	2010-09-15 10:42:11.000000000 -0700
@@ -344,10 +344,6 @@
 use Config;
 use strict;
 
-"\$Config{'archname'}-\$Config{'osvers'}" eq
-"$Config{'archname'}-$Config{'osvers'}" or
-	die "Errno architecture ($Config{'archname'}-$Config{'osvers'}) does not match executable architecture (\$Config{'archname'}-\$Config{'osvers'})";
-
 \$VERSION = "$VERSION";
 \$VERSION = eval \$VERSION;
 \@ISA = qw(Exporter);
