diff -x '*~' -x '.*' --speed-large-files --minimal -Nru tmp/mms-1.1.0-1788/plugins/feature/media-rss/media_rss.cpp work/main.d/mms-1.1.0-1788/plugins/feature/media-rss/media_rss.cpp
--- tmp/mms-1.1.0-1788/plugins/feature/media-rss/media_rss.cpp	2010-03-06 12:48:51.000000000 -0300
+++ new-mms-1.1.0-1788/plugins/feature/media-rss/media_rss.cpp	2010-05-03 19:45:49.000000000 -0300
@@ -507,9 +516,10 @@
   int curlayer = render->device->get_current_layer();
   render->device->switch_to_layer(curlayer+1);
   render->device->reset_layout_attribs_nowait();
-  render->device->animation_zoom(3,3,1,1,40,curlayer+1);
-  render->device->animation_fade(0,1,50,curlayer+1);
-  render->device->animation_fade(1,0,50,curlayer);
+  render->device->set_layout_alpha(0.0, curlayer+1);
+  render->device->animation_fade(0,1,60,curlayer+1);
+  render->device->animation_zoom(1,2,1,1,40,curlayer+1);
+  render->device->animation_fade(1,0,20,curlayer);
   render->device->animation_section_end(); 
 
   exit_loop = false;
@@ -578,9 +588,9 @@
   render->device->animation_section_begin();
   render->device->switch_to_layer(curlayer);
   render->device->reset_layout_attribs_nowait();
-  render->device->animation_fade(0,1,80,curlayer);
-  render->device->animation_zoom(1,1,4,4,40,curlayer+1);
-  render->device->animation_fade(1,0,40,curlayer+1);
+  render->device->animation_fade(0,1,40,curlayer);
+  render->device->animation_zoom(1,1,1,2,60,curlayer+1);
+  render->device->animation_fade(1,0,30,curlayer+1);
   render->device->animation_section_end();
 
   //input_master->restore_map();
