diff -Naur mysql-5.1.58-old/extra/Makefile.am mysql-5.1.58-new/extra/Makefile.am
--- mysql-5.1.58-old/extra/Makefile.am	2011-07-07 19:21:34.000000000 -0700
+++ mysql-5.1.58-new/extra/Makefile.am	2011-07-07 19:22:20.000000000 -0700
@@ -32,7 +32,7 @@
 $(top_builddir)/include/mysqld_error.h: comp_err.c \
 			$(top_srcdir)/sql/share/errmsg.txt
 	$(MAKE) $(AM_MAKEFLAGS) comp_err$(EXEEXT)
-	$(top_builddir)/extra/comp_err$(EXEEXT) \
+	mysql-comp_err \
 	--charset=$(top_srcdir)/sql/share/charsets \
 	--out-dir=$(top_builddir)/sql/share/ \
 	--header_file=$(top_builddir)/include/mysqld_error.h \
diff -Naur mysql-5.1.58-old/extra/Makefile.in mysql-5.1.58-new/extra/Makefile.in
--- mysql-5.1.58-old/extra/Makefile.in	2011-07-07 19:21:34.000000000 -0700
+++ mysql-5.1.58-new/extra/Makefile.in	2011-07-07 19:22:20.000000000 -0700
@@ -937,7 +937,7 @@
 $(top_builddir)/include/mysqld_error.h: comp_err.c \
 			$(top_srcdir)/sql/share/errmsg.txt
 	$(MAKE) $(AM_MAKEFLAGS) comp_err$(EXEEXT)
-	$(top_builddir)/extra/comp_err$(EXEEXT) \
+	mysql-comp_err$(EXEEXT) \
 	--charset=$(top_srcdir)/sql/share/charsets \
 	--out-dir=$(top_builddir)/sql/share/ \
 	--header_file=$(top_builddir)/include/mysqld_error.h \
diff -Naur mysql-5.1.58-old/sql/Makefile.am mysql-5.1.58-new/sql/Makefile.am
--- mysql-5.1.58-old/sql/Makefile.am	2011-07-07 19:21:34.000000000 -0700
+++ mysql-5.1.58-new/sql/Makefile.am	2011-07-07 19:22:20.000000000 -0700
@@ -175,7 +175,7 @@
 # this avoid the rebuild of the built files in a source dist
 lex_hash.h:	gen_lex_hash.cc lex.h
 		$(MAKE) $(AM_MAKEFLAGS) gen_lex_hash$(EXEEXT)
-		./gen_lex_hash$(EXEEXT) > $@-t
+		mysql-gen_lex_hash > $@-t
 		$(MV) $@-t $@
 
 # For testing of udf_example.so
diff -Naur mysql-5.1.58-old/sql/Makefile.in mysql-5.1.58-new/sql/Makefile.in
--- mysql-5.1.58-old/sql/Makefile.in	2011-07-07 19:21:34.000000000 -0700
+++ mysql-5.1.58-new/sql/Makefile.in	2011-07-07 19:22:20.000000000 -0700
@@ -1305,7 +1305,7 @@
 # this avoid the rebuild of the built files in a source dist
 lex_hash.h:	gen_lex_hash.cc lex.h
 		$(MAKE) $(AM_MAKEFLAGS) gen_lex_hash$(EXEEXT)
-		./gen_lex_hash$(EXEEXT) > $@-t
+		mysql-gen_lex_hash > $@-t
 		$(MV) $@-t $@
 
 # We might have some stuff not built in this build, but that we want to install
