diff -Naur xf86-video-nouveau-b795ca6e97fae9735843748585401098dae1c3e1-old/configure.ac xf86-video-nouveau-b795ca6e97fae9735843748585401098dae1c3e1-new/configure.ac
--- xf86-video-nouveau-b795ca6e97fae9735843748585401098dae1c3e1-old/configure.ac	2011-01-14 07:28:34.000000000 -0800
+++ xf86-video-nouveau-b795ca6e97fae9735843748585401098dae1c3e1-new/configure.ac	2011-01-14 07:30:40.000000000 -0800
@@ -76,7 +76,8 @@
 		  HAVE_XEXTPROTO_71="yes"; AC_DEFINE(HAVE_XEXTPROTO_71, 1, [xextproto 7.1 available]),
 		  HAVE_XEXTPROTO_71="no")
 AM_CONDITIONAL(HAVE_XEXTPROTO_71, [ test "$HAVE_XEXTPROTO_71" = "yes" ])
-sdkdir=$(pkg-config --variable=sdkdir xorg-server)
+m4_pattern_allow([^PKG_CONFIG_SYSROOT_DIR$])
+sdkdir="$PKG_CONFIG_SYSROOT_DIR$($PKG_CONFIG --variable=sdkdir xorg-server)"
 
 # Checks for libraries.
 PKG_CHECK_MODULES([PCIACCESS], [pciaccess >= 0.10])
