diff -Naur xf86-input-evdev-2.6.0-old/configure.ac xf86-input-evdev-2.6.0-new/configure.ac
--- xf86-input-evdev-2.6.0-old/configure.ac	2011-01-10 18:46:36.000000000 -0800
+++ xf86-input-evdev-2.6.0-new/configure.ac	2011-01-10 21:16:11.000000000 -0800
@@ -58,7 +58,8 @@
 
 # X Server SDK location is required to install evdev header files
 # This location is also relayed in the xorg-evdev.pc file
-sdkdir=`$PKG_CONFIG --variable=sdkdir xorg-server`
+m4_pattern_allow([^PKG_CONFIG_SYSROOT_DIR$])
+sdkdir="$PKG_CONFIG_SYSROOT_DIR$($PKG_CONFIG --variable=sdkdir xorg-server)"
 
 # Workaround overriding sdkdir to be able to create a tarball when user has no
 # write permission in sdkdir. See DISTCHECK_CONFIGURE_FLAGS in Makefile.am
