diff -Naur libdvdread-1217-old/misc/dvdread-config.in libdvdread-1217-new/misc/dvdread-config.in
--- libdvdread-1217-old/misc/dvdread-config.in	2010-08-10 09:11:10.000000000 -0700
+++ libdvdread-1217-new/misc/dvdread-config.in	2010-08-10 09:12:04.000000000 -0700
@@ -72,9 +72,9 @@
 fi
 
 if test "$echo_cflags" = "yes"; then
-      echo -I@includedir@
+      echo -I$DESTDIR@includedir@
 fi
 
 if test "$echo_libs" = "yes"; then
-      echo -L@libdir@ -ldvdread
+      echo -L$DESTDIR@libdir@ -ldvdread
 fi
