diff -Naur NVIDIA-Linux-x86-173.14.30-pkg1-old/usr/src/nv/conftest.sh NVIDIA-Linux-x86-173.14.30-pkg1-new/usr/src/nv/conftest.sh
--- NVIDIA-Linux-x86-173.14.30-pkg1-old/usr/src/nv/conftest.sh	2011-04-14 09:26:35.000000000 -0700
+++ NVIDIA-Linux-x86-173.14.30-pkg1-new/usr/src/nv/conftest.sh	2011-07-22 16:14:03.000000000 -0700
@@ -77,7 +77,7 @@
 build_cflags() {
     ARCH=`uname -m | sed -e 's/i.86/i386/'`
 
-    BASE_CFLAGS="-D__KERNEL__ \
+    BASE_CFLAGS="-O2 -D__KERNEL__ \
 -DKBUILD_BASENAME=\"#conftest$$\" -DKBUILD_MODNAME=\"#conftest$$\" \
 -nostdinc -isystem $ISYSTEM"
 
@@ -1699,7 +1699,8 @@
         # Run a series of compile tests to determine the set of interfaces
         # and features available in the target kernel.
         #
-        FILES="linux/semaphore.h generated/autoconf.h"
+        FILES="linux/semaphore.h generated/autoconf.h generated/compile.h"
+        FILES="$FILES generated/utsrelease.h"
         shift 5
 
         rm -f conftest.h
diff -Naur NVIDIA-Linux-x86-173.14.30-pkg1-old/usr/src/nv/nv-linux.h NVIDIA-Linux-x86-173.14.30-pkg1-new/usr/src/nv/nv-linux.h
--- NVIDIA-Linux-x86-173.14.30-pkg1-old/usr/src/nv/nv-linux.h	2011-04-14 09:26:35.000000000 -0700
+++ NVIDIA-Linux-x86-173.14.30-pkg1-new/usr/src/nv/nv-linux.h	2011-07-22 16:14:03.000000000 -0700
@@ -32,6 +32,8 @@
 #  error This driver does not support 2.5 kernels!
 #elif LINUX_VERSION_CODE < KERNEL_VERSION(2, 7, 0)
 #  define KERNEL_2_6
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(3, 0, 0)
+#  define KERNEL_3
 #else
 #  error This driver does not support development kernels!
 #endif
