diff -Naur xscreensaver-5.14-old/Makefile.in xscreensaver-5.14-new/Makefile.in
--- xscreensaver-5.14-old/Makefile.in	2011-05-21 07:02:03.000000000 -0700
+++ xscreensaver-5.14-new/Makefile.in	2011-05-21 07:03:11.000000000 -0700
@@ -6,7 +6,7 @@
 VPATH		= @srcdir@
 
 SHELL		= /bin/sh
-SUBDIRS 	= utils driver hacks hacks/glx po
+SUBDIRS 	= utils driver hacks hacks/glx
 #SUBDIRS 	= utils driver hacks hacks/glx
 SUBDIRS2	= $(SUBDIRS) OSX
 TARFILES	= README README.hacking README.VMS INSTALL \
