--- old-xbmc-11.0/configure.in	2012-07-29 22:04:01.976549343 -0300
+++ new-xbmc-11.0/configure.in	2012-06-12 14:00:36.760667000 -0300
@@ -2,7 +2,7 @@
 # Process this file with autoconf to produce a configure script.
 
 AC_PREREQ(2.59)
-AC_INIT([xbmc], [11.0], [http://trac.xbmc.org])
+AC_INIT([castorbox HD], [HD], [http://trac.xbmc.org])
 AC_CONFIG_HEADERS([xbmc/config.h])
 AH_TOP([#pragma once])
 m4_include([m4/ax_python_devel.m4])

