diff -x '*~' -x '.*' -x '*.orig' -x '*.rej' --speed-large-files --minimal -Nru tmp/xbmc-11.0/configure.in work/main.d/xbmc-11.0/configure.in
--- tmp/xbmc-11.0/configure.in	2012-03-21 19:07:50.000000000 -0300
+++ new-xbmc-11.0/configure.in	2012-06-09 16:36:32.560666123 -0300
@@ -20,10 +20,16 @@
 [
   if [[ "$host_vendor" != "apple" ]]; then
     AC_MSG_CHECKING([for lib$2 soname])
-    $1_FILENAME=$($CC -nostdlib -o /dev/null $LDFLAGS -l$2 -Wl,-M 2>/dev/null | grep "^LOAD.*$2" | awk '{V=2; print $V}')
-    if [[ ! -z $$1_FILENAME ]]; then
-      $1_SONAME=$(objdump -p $$1_FILENAME | grep "SONAME.*$2" | awk '{V=2; print $V}')
-    fi
+        $1_SONAME=$( $CC -print-file-name=lib$2.so | \
+        while read output; do objdump -p $output | \
+        grep "SONAME" | \
+        sed -e 's/  *SONAME  *//'; done 2> /dev/null )
+
+#    AC_MSG_CHECKING([for lib$2 soname])
+#    $1_FILENAME=$($CC -nostdlib -o /dev/null $LDFLAGS -l$2 -Wl,-M 2>/dev/null | grep "^LOAD.*$2" | awk '{V=2; print $V}')
+#    if [[ ! -z $$1_FILENAME ]]; then
+#      $1_SONAME=$(objdump -p $$1_FILENAME | grep "SONAME.*$2" | awk '{V=2; print $V}')
+#    fi
   else
     AC_MSG_CHECKING([for lib$2 dylib])
     gcc_lib_path=[`$CC -print-search-dirs 2>/dev/null | fgrep libraries: | sed 's/[^=]*=\(.*\)/\1/' | sed 's/:/ /g'`]
