diff -x '*~' -x '.*' --speed-large-files --minimal -Nru tmp/resolvconf-1.42/bin/list-records new-resolvconf-1.42/bin/list-records
--- tmp/resolvconf-1.42/bin/list-records	2008-06-05 05:50:40.000000000 -0300
+++ new-resolvconf-1.42/bin/list-records	2009-07-16 09:34:56.000000000 -0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 # Need bash because we use nullglob, extglob
 #
 # list-records
@@ -36,7 +36,7 @@
 		/^$/d
 		' /etc/resolvconf/interface-order)"
 fi
-shopt -s nullglob extglob
+#DISABLED shopt -s nullglob extglob
 # Pathname expansion occurs on the following line resulting, in general,
 # in multiple instances of filenames; duplicates must be removed.
 uniquify $PATTERNS
diff -x '*~' -x '.*' --speed-large-files --minimal -Nru tmp/resolvconf-1.42/bin/resolvconf new-resolvconf-1.42/bin/resolvconf
--- tmp/resolvconf-1.42/bin/resolvconf	2008-06-05 05:50:40.000000000 -0300
+++ new-resolvconf-1.42/bin/resolvconf	2009-07-16 09:34:56.000000000 -0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 # Need bash because of use of ${FOO/bar}
 #
 # Licensed under the GNU GPL.  See /usr/share/doc/resolvconf/copyright.
diff -x '*~' -x '.*' --speed-large-files --minimal -Nru tmp/resolvconf-1.42/etc/resolvconf/update.d/bind new-resolvconf-1.42/etc/resolvconf/update.d/bind
--- tmp/resolvconf-1.42/etc/resolvconf/update.d/bind	2008-06-05 05:50:40.000000000 -0300
+++ new-resolvconf-1.42/etc/resolvconf/update.d/bind	2009-07-16 09:34:56.000000000 -0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 # Need bash because we use ${foo//bar/baz}
 #
 # Script to update the named options file
