diff -Naur vlc-1.0.1-old/bin/vlc.c vlc-1.0.1-new/bin/vlc.c
--- vlc-1.0.1-old/bin/vlc.c	2009-07-28 03:22:41.000000000 -0700
+++ vlc-1.0.1-new/bin/vlc.c	2009-07-28 03:24:11.000000000 -0700
@@ -151,7 +151,6 @@
             libvlc_exception_clear (&ex);
             pthread_sigmask (SIG_UNBLOCK, &set, NULL);
         }
-        libvlc_add_intf (vlc, "globalhotkeys,none", &ex);
         libvlc_add_intf (vlc, NULL, &ex);
         libvlc_playlist_play (vlc, -1, 0, NULL, &dummy);
         libvlc_wait (vlc);
