diff -Naur pkg-config-0.25-old/pkg.m4 pkg-config-0.25-new/pkg.m4
--- pkg-config-0.25-old/pkg.m4	2009-12-06 12:39:02.000000000 -0800
+++ pkg-config-0.25-new/pkg.m4	2010-08-10 14:34:03.000000000 -0700
@@ -135,7 +135,7 @@
 Consider adjusting the PKG_CONFIG_PATH environment variable if you
 installed software in a non-standard prefix.
 
-_PKG_TEXT])dnl
+_PKG_TEXT])[]dnl
         ])
 elif test $pkg_failed = untried; then
      	AC_MSG_RESULT([no])
