diff --speed-large-files --minimal -Nru tmp/imms-3.1.0-rc2/rules.mk work/main.d/imms-3.1.0-rc2/rules.mk
--- tmp/imms-3.1.0-rc2/rules.mk	2007-01-17 12:53:04.000000000 -0300
+++ imms-3.1.0-rc2/rules.mk	2007-12-20 15:43:46.000000000 -0300
@@ -31,4 +31,4 @@
 install: all plugins_install programs_install
 
 programs_install: $(patsubst %,%_install,$(OPTIONAL))
-	${INSTALL} -D immsd immstool $(bindir)
+	${INSTALL} -D immsd immstool $(DESTDIR)$(bindir)
