diff -Naur glibc-2.16.0-old/Makefile.in glibc-2.16.0-new/Makefile.in
--- glibc-2.16.0-old/Makefile.in	2012-07-01 17:34:12.000000000 -0700
+++ glibc-2.16.0-new/Makefile.in	2012-07-01 17:34:42.000000000 -0700
@@ -12,6 +12,9 @@
 locale: lib
 	$(MAKE) -r PARALLELMFLAGS="$(PARALLELMFLAGS)" CVSOPTS="$(CVSOPTS)" -C $(srcdir)/locale objdir=`pwd`
 
+sunrpc: lib
+	$(MAKE) -r PARALLELMFLAGS="$(PARALLELMFLAGS)" CVSOPTS="$(CVSOPTS)" -C $(srcdir)/sunrpc objdir=`pwd`
+
 install:
 	LANGUAGE=C LC_ALL=C; export LANGUAGE LC_ALL; \
 	$(MAKE) -r PARALLELMFLAGS="$(PARALLELMFLAGS)" -C $(srcdir) objdir=`pwd` $@
