diff -Naur glibc-2.16.0-old/elf/ldd.bash.in glibc-2.16.0-new/elf/ldd.bash.in
--- glibc-2.16.0-old/elf/ldd.bash.in	2012-07-01 17:38:45.000000000 -0700
+++ glibc-2.16.0-new/elf/ldd.bash.in	2012-07-01 17:39:59.000000000 -0700
@@ -111,7 +111,7 @@
 # environments where the executed program might not have permissions
 # to write to the console/tty.  But only bash 3.x supports the pipefail
 # option, and we don't bother to handle the case for older bash versions.
-if set -o pipefail 2> /dev/null; then
+if `set -o pipefail 2> /dev/null ; echo $?`; then
   try_trace() {
     eval $add_env '"$@"' | cat
   }
