diff -Naur transcode-1.1.5-old/configure.in transcode-1.1.5-new/configure.in
--- transcode-1.1.5-old/configure.in	2009-10-31 09:39:02.000000000 -0700
+++ transcode-1.1.5-new/configure.in	2009-11-01 09:35:25.000000000 -0800
@@ -872,8 +872,9 @@
     [lame_version="`./conftest$ac_exeext`"],
     [AC_MSG_RESULT([failed])
      TC_PKG_ERROR(lame, lame.h, yes, lame, [http://www.mp3dev.org/],
-      [cannot compile and run a test program])],,
-    [AC_MSG_RESULT([cross compiling; assumed OK...])])
+      [cannot compile and run a test program])],
+    [echo $ac_n "cross compiling; assumed OK... $ac_c"
+    lame_version=398])
   CFLAGS="$ac_save_CFLAGS"
   LIBS="$ac_save_LIBS"
 
