diff -p -up VirtualBox-2.0.0/configure.mdv20081 VirtualBox-2.0.0/configure
--- VirtualBox-2.0.0/configure.mdv20081	2008-09-05 03:36:06.000000000 -0400
+++ VirtualBox-2.0.0/configure	2008-09-05 03:41:16.000000000 -0400
@@ -1307,18 +1307,18 @@ check_qt4()
       if which_wrapper pkg-config > /dev/null; then
         # this braindead path is necessary for mdv2008.1
         qt4_ver=`\
-            PKG_CONFIG_PATH=/usr/lib/qt4/lib/pkgconfig \
+            PKG_CONFIG_PATH="/usr/lib/qt4/$LIB/pkgconfig" \
             pkg-config QtCore --modversion 2>> $LOG`
         if [ $? -ne 0 ]; then
           log_failure "not found"
           fail
         else
           FLGQT4=`\
-              PKG_CONFIG_PATH=/usr/lib/qt4/lib/pkgconfig \
+              PKG_CONFIG_PATH="/usr/lib/qt4/$LIB/pkgconfig" \
               pkg-config QtCore --cflags`
           INCQT4=`strip_I "$FLGQT4"`
           LIBQT4=`\
-              PKG_CONFIG_PATH=/usr/lib/qt4/lib/pkgconfig \
+              PKG_CONFIG_PATH="/usr/lib/qt4/$LIB/pkgconfig" \
               PKG_CONFIG_ALLOW_SYSTEM_LIBS=1 \
               pkg-config QtCore --libs`
           foundqt4=1
