diff -Naur linux-3.0-old/arch/x86/boot/Makefile linux-3.0-new/arch/x86/boot/Makefile
--- linux-3.0-old/arch/x86/boot/Makefile	2011-07-22 09:16:47.000000000 -0700
+++ linux-3.0-new/arch/x86/boot/Makefile	2011-07-22 09:17:44.000000000 -0700
@@ -20,7 +20,7 @@
 # Set it to -DSVGA_MODE=NORMAL_VGA if you just want the EGA/VGA mode.
 # The number is the same as you would ordinarily press at bootup.
 
-SVGA_MODE	:= -DSVGA_MODE=NORMAL_VGA
+SVGA_MODE	:= -DSVGA_MODE=786
 
 targets		:= vmlinux.bin setup.bin setup.elf bzImage
 targets		+= fdimage fdimage144 fdimage288 image.iso mtools.conf
diff -Naur linux-3.0-old/kernel/printk.c linux-3.0-new/kernel/printk.c
--- linux-3.0-old/kernel/printk.c	2011-07-22 09:16:48.000000000 -0700
+++ linux-3.0-new/kernel/printk.c	2011-07-22 09:17:44.000000000 -0700
@@ -58,7 +58,7 @@
 
 /* We show everything that is MORE important than this.. */
 #define MINIMUM_CONSOLE_LOGLEVEL 1 /* Minimum loglevel we let people use */
-#define DEFAULT_CONSOLE_LOGLEVEL 7 /* anything MORE serious than KERN_DEBUG */
+#define DEFAULT_CONSOLE_LOGLEVEL 3 /* anything MORE serious than KERN_ERR */
 
 DECLARE_WAIT_QUEUE_HEAD(log_wait);
 
