diff -Naur lirc-0.8.5pre2-old/configure.ac lirc-0.8.5pre2-new/configure.ac
--- lirc-0.8.5pre2-old/configure.ac	2009-03-31 19:03:37.000000000 -0700
+++ lirc-0.8.5pre2-new/configure.ac	2009-03-31 19:05:28.000000000 -0700
@@ -22,8 +22,8 @@
 AC_PATH_PROG(LIBUSB_CONFIG, libusb-config)
 AC_PROG_LN_S
 AC_PROG_LIBTOOL
-AM_PATH_PYTHON
-AM_CONDITIONAL([HAVE_PYTHON], [test "$PYTHON" != ""])
+AM_PATH_PYTHON(,,[:])
+AM_CONDITIONAL([HAVE_PYTHON], [test "$PYTHON" != :])
 
 dnl Checks for header files.
 AC_HEADER_STDC
