diff --speed-large-files --minimal -Nru tmp/lighttpd-1.4.18/configure.in work/main.d/lighttpd-1.4.18/configure.in
--- lighttpd-1.4.18/configure.in	2007-08-28 21:49:56.000000000 -0300
+++ lighttpd-1.4.18/configure.in	2007-12-04 18:36:14.000000000 -0300
@@ -251,7 +251,7 @@
     [WITH_PCRE=$withval],[WITH_PCRE=yes])
 AC_MSG_RESULT([$WITH_PCRE])
 
-if test "x$cross_compiling" = xno -a "$WITH_PCRE" != "no"; then
+if test "$WITH_PCRE" != "no"; then
   AC_PATH_PROG(PCRECONFIG, pcre-config)
 
   if test x"$PCRECONFIG" != x; then
