diff -Naur Mesa-7.9-old/configure.ac Mesa-7.9-new/configure.ac
--- Mesa-7.9-old/configure.ac	2010-10-04 18:58:00.000000000 -0700
+++ Mesa-7.9-new/configure.ac	2010-10-30 16:13:21.000000000 -0700
@@ -4,9 +4,9 @@
 
 dnl Versioning - scrape the version from configs/default
 m4_define([mesa_version],
-    [m4_esyscmd([${MAKE-make} -s -f bin/version.mk version | tr -d '\n' | tr -d '\r'])])
+    [7.9.0])
 m4_ifval(mesa_version,,
-    [m4_fatal([Failed to get the Mesa version from `make -f bin/version.mk version`])])
+    [m4_fatal([Failed to get the Mesa version from `make -f bin/version.mk version-mk-version`])])
 
 dnl Tell the user about autoconf.html in the --help output
 m4_divert_once([HELP_END], [
