diff -Naur sysvinit-2.86-old/src/halt.c sysvinit-2.86-new/src/halt.c
--- sysvinit-2.86-old/src/halt.c	2004-07-30 05:16:18.000000000 -0700
+++ sysvinit-2.86-new/src/halt.c	2005-12-02 17:50:31.000000000 -0800
@@ -234,8 +234,19 @@
 		 *	See if we are in runlevel 0 or 6.
 		 */
 		c = get_runlevel();
-		if (c != '0' && c != '6')
+		if (c != '0' && c != '6') {
+			char *file;
+
+			if (do_poweroff) {
+				file = strdup("/poweroff");
+			} else {
+				file = strdup("/halt");
+			}
+			close(open(file, O_CREAT|O_RDWR, 0644));
+			free(file);
+
 			do_shutdown(do_reboot ? "-r" : "-h", tm);
+		}
 	}
 
 	/*
