diff -Naur sysvinit-2.86-old/src/init.c sysvinit-2.86-new/src/init.c
--- sysvinit-2.86-old/src/init.c	2004-07-30 05:16:20.000000000 -0700
+++ sysvinit-2.86-new/src/init.c	2009-05-19 09:11:04.000000000 -0700
@@ -2213,7 +2213,7 @@
 			break;
 		default:
 			if (warn)
-			  initlog(L_VB,
+			  initlog(L_SY,
 				"no more processes left in this runlevel");
 			warn = 0;
 			loglevel = -1;
@@ -2222,7 +2222,7 @@
 			break;
 	}
 	if (loglevel > 0) {
-		initlog(L_VB, "Entering runlevel: %c", runlevel);
+		initlog(L_SY, "Entering runlevel: %c", runlevel);
 		write_utmp_wtmp("runlevel", "~~", runlevel + 256 * oldlevel, RUN_LVL, "~");
 		thislevel = runlevel;
 		prevlevel = oldlevel;
@@ -2423,7 +2423,7 @@
   	/*
 	 *	Say hello to the world
 	 */
-  	initlog(L_CO, bootmsg, "booting");
+  	initlog(L_SY, bootmsg, "booting");
 
   	/*
 	 *	See if we have to start an emergency shell.
@@ -2448,7 +2448,7 @@
 	/*
 	 *	Restart: unblock signals and let the show go on
 	 */
-	initlog(L_CO, bootmsg, "reloading");
+	initlog(L_SY, bootmsg, "reloading");
 	sigfillset(&sgt);
 	sigprocmask(SIG_UNBLOCK, &sgt, NULL);
   }
