diff -Naur libcdio-0.82-old/configure.ac libcdio-0.82-new/configure.ac
--- libcdio-0.82-old/configure.ac	2009-10-27 04:26:32.000000000 -0700
+++ libcdio-0.82-new/configure.ac	2009-10-27 07:18:42.000000000 -0700
@@ -86,7 +86,7 @@
 AM_CONDITIONAL([ENABLE_CXX_BINDINGS], [test "x$enable_cxx" != "xno"])
 
 AC_ARG_ENABLE(cpp-progs,
-	[  --enable-cpp-progs      make C++ example programs (default enabled)])
+	AC_HELP_STRING([--disable-cpp-progs], [make C++ example programs (default enabled)]))
 AM_CONDITIONAL(ENABLE_CPP, test x"$enable_cpp_progs" = "xyes")
 
 AC_ARG_ENABLE(example-progs,
@@ -543,14 +543,17 @@
 fi
 AC_SUBST(HAVE_JOLIET)
 
-if test x$enable_rock = x; then
-  enable_rock=yes
-  AC_ARG_ENABLE(rock,
-	[  --enable-rock           include Rock-Ridge extension support (default enabled)],
-	enable_rock=yes, enable_rock=no)
-fi
+AC_ARG_ENABLE(rock,
+	AC_HELP_STRING([--disable-rock], [include Rock-Ridge extension support (default enabled)]),
+	[
+		AS_CASE(["$enableval"],
+			[yes], [enable_rock="yes"],
+			[no],  [enable_rock="no"],
+			       [AC_MSG_ERROR([bad value '${enableval}' for --enable-rock])])
+	],
+	enable_rock="yes")
 
-if test "${enable_rock}" != "no" ; then
+if test "x$enable_rock" != "xno" ; then
    AC_DEFINE(HAVE_ROCK, [1], 
    [Define 1 if you want ISO-9660 Rock-Ridge extension support.])
    HAVE_ROCK=1
@@ -558,13 +561,16 @@
 AC_SUBST(HAVE_ROCK)
 AM_CONDITIONAL(ENABLE_ROCK, test x"$enable_rock" = "xyes")
 
-if test x$enable_cddb = x; then
-  enable_cddb=yes
-  AC_ARG_ENABLE(cddb,
-	[  --enable-cddb           include CDDB lookups in cd_info (default enabled)],	
-	enable_cddb=yes, enable_cddb=no)
-fi
-if test x$enable_cddb != "no" ; then
+AC_ARG_ENABLE(cddb,
+	AC_HELP_STRING([--disable-cddb], [include CDDB lookups in cd_info (default enabled)]),
+	[
+		AS_CASE(["$enableval"],
+			[yes], [enable_cddb="yes"],
+			[no],  [enable_cddb="no"],
+			       [AC_MSG_ERROR([bad value '${enableval}' for --enable-cddb])])
+	],
+	enable_cddb="yes")
+if test "x$enable_cddb" != "xno" ; then
    PKG_CHECK_MODULES(CDDB, libcddb >= 1.0.1, [
     HAVE_CDDB=yes 
     AC_DEFINE(HAVE_CDDB, [], [Define this if you have libcddb installed])
@@ -594,13 +600,16 @@
 AM_CONDITIONAL(BUILD_CDDA_PLAYER, test "x$enable_cdda_player" = "xyes")
 AC_SUBST(CDDA_PLAYER_LIBS)
 
-if test x$enable_vcd_info = x; then
 AC_ARG_ENABLE(vcd_info,
-	[  --enable-vcd-info      include Video CD Info from libvcd],
-		enable_vcd_info=yes,
-		enable_vcd_info=no)
-fi
-if test x$enable_vcd_info = xyes; then
+	AC_HELP_STRING([--enable-vcd-info], [include Video CD Info from libvcd (default disabled)]),
+	[
+		AS_CASE(["$enableval"],
+			[yes], [enable_vcd_info="yes"],
+			[no],  [enable_vcd_info="no"],
+			       [AC_MSG_ERROR([bad value '${enableval}' for --enable-vcd-info])])
+	],
+	enable_vcd_info="no")
+if test "x$enable_vcd_info" = "xyes" ; then
   PKG_CHECK_MODULES(VCDINFO, libvcdinfo >= 0.7.21, 
   [AC_DEFINE([HAVE_VCDINFO],1, 
   [Define this if you have libvcdinfo installed])], 
