diff -Naur libX11-1.4.0-old/configure.ac libX11-1.4.0-new/configure.ac
--- libX11-1.4.0-old/configure.ac	2010-11-21 14:02:48.000000000 -0800
+++ libX11-1.4.0-new/configure.ac	2010-11-21 17:05:39.000000000 -0800
@@ -36,6 +36,7 @@
 AC_PROG_LIBTOOL
 AC_PROG_CC
 PKG_PROG_PKG_CONFIG
+m4_pattern_allow([^PKG_CONFIG_SYSROOT_DIR$])
 
 if test x"$CC_FOR_BUILD" = x; then
 	if test x"$cross_compiling" = xyes; then
@@ -313,7 +314,7 @@
 # Find keysymdef.h
 #
 AC_MSG_CHECKING([keysym definitions])
-KEYSYMDEFDIR=`$PKG_CONFIG --variable=includedir xproto`/X11
+KEYSYMDEFDIR=$PKG_CONFIG_SYSROOT_DIR`$PKG_CONFIG --variable=includedir xproto`/X11
 FILES="keysymdef.h XF86keysym.h Sunkeysym.h DECkeysym.h HPkeysym.h"
 for i in $FILES; do
     if test -f "$KEYSYMDEFDIR/$i"; then
