diff -Naur Python-2.7-old/setup.py Python-2.7-new/setup.py
--- Python-2.7-old/setup.py	2010-06-27 05:36:16.000000000 -0700
+++ Python-2.7-new/setup.py	2010-07-07 07:26:09.000000000 -0700
@@ -222,15 +222,19 @@
             if ext.name in remove_modules:
                 self.extensions.remove(ext)
 
-        # When you run "make CC=altcc" or something similar, you really want
-        # those environment variables passed into the setup.py phase.  Here's
-        # a small set of useful ones.
-        compiler = os.environ.get('CC')
         args = {}
         # unfortunately, distutils doesn't let us provide separate C and C++
         # compilers
+        (compiler,ccshared,cflags) = sysconfig.get_config_vars('CC','CCSHARED','CFLAGS')
+        # When you run "make CC=altcc" or something similar, you really want
+        # those environment variables passed into the setup.py phase.
+        if 'CC' in os.environ:
+            compiler = os.environ['CC']
+        if 'CCSHARED' in os.environ:
+            ccshared = os.environ['CCSHARED']
+        if 'CFLAGS' in os.environ:
+            cflags = os.environ['CFLAGS']
         if compiler is not None:
-            (ccshared,cflags) = sysconfig.get_config_vars('CCSHARED','CFLAGS')
             args['compiler_so'] = compiler + ' ' + ccshared + ' ' + cflags
         self.compiler.set_executables(**args)
 
