diff -Naur vdpau-video-0.6.0-old/src/Makefile.am vdpau-video-0.6.0-new/src/Makefile.am
--- vdpau-video-0.6.0-old/src/Makefile.am	2009-11-27 02:36:49.000000000 -0800
+++ vdpau-video-0.6.0-new/src/Makefile.am	2009-11-27 22:13:05.000000000 -0800
@@ -67,6 +67,7 @@
 install-data-hook:
 	cd $(DESTDIR)$(LIBVA_DRIVERS_PATH) ;			\
 	for drv in $(DRIVERS); do				\
+	    rm -f $${drv}_drv_video.so;                         \
 	    ln -s vdpau_drv_video.so $${drv}_drv_video.so;	\
 	done
 
diff -Naur vdpau-video-0.6.0-old/src/Makefile.in vdpau-video-0.6.0-new/src/Makefile.in
--- vdpau-video-0.6.0-old/src/Makefile.in	2009-11-27 02:39:59.000000000 -0800
+++ vdpau-video-0.6.0-new/src/Makefile.in	2009-11-27 22:13:05.000000000 -0800
@@ -583,6 +583,7 @@
 install-data-hook:
 	cd $(DESTDIR)$(LIBVA_DRIVERS_PATH) ;			\
 	for drv in $(DRIVERS); do				\
+	    rm -f $${drv}_drv_video.so;                         \
 	    ln -s vdpau_drv_video.so $${drv}_drv_video.so;	\
 	done
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
