diff -Naur nss-3.12.10-old/mozilla/security/nss/cmd/shlibsign/sign.sh nss-3.12.10-new/mozilla/security/nss/cmd/shlibsign/sign.sh
--- nss-3.12.10-old/mozilla/security/nss/cmd/shlibsign/sign.sh	2011-05-10 18:04:15.000000000 -0700
+++ nss-3.12.10-new/mozilla/security/nss/cmd/shlibsign/sign.sh	2011-05-10 18:04:57.000000000 -0700
@@ -18,8 +18,8 @@
         PATH=${ARG1}/lib:${ARG1}/bin:${ARG4}:${PATH}
     fi
     export PATH
-    echo "${2}"/shlibsign -v -i "${5}"
-    "${2}"/shlibsign -v -i "${5}"
+    echo nss-shlibsign -v -i "${5}"
+    nss-shlibsign -v -i "${5}"
     ;;
 *)
     LIBPATH=`(cd "${1}"/lib; pwd)`:`(cd "${4}"; pwd)`:$LIBPATH
@@ -34,7 +34,7 @@
     export LIBRARY_PATH
     ADDON_PATH=${1}/lib:${4}:$ADDON_PATH
     export ADDON_PATH
-    echo "${2}"/shlibsign -v -i "${5}"
-    "${2}"/shlibsign -v -i "${5}"
+    echo nss-shlibsign -v -i "${5}"
+    nss-shlibsign -v -i "${5}"
     ;;
 esac
